lib/src/data/repository/genre_repository_impl.dart

```diff
@@ -1,34 +1,53 @@
 import 'dart:core';
-import '../../core/util/api_constants.dart';
+import '../../core/util/constants.dart';
 import '../../domain/entity/genre.dart';
 import '../../domain/repository/i_repository.dart';
+import '../datasource/local/DAOs/genre_dao.dart';
+import '../datasource/local/movie_database.dart';
 import '../datasource/remote/i_api_service.dart';
 import '../../core/util/data_state.dart';
 import '../model/genre_model.dart';
 
 class GenreRepositoryImpl implements IRepository<DataState<List<Genre>>> {
   IApiService genresService;
+  Future<AppDatabase> databaseInstance;
 
   GenreRepositoryImpl({
     required this.genresService,
+    required this.databaseInstance,
   });
 
   @override
   Future<DataState<List<Genre>>> getData() async {
+    AppDatabase database = await databaseInstance;
```

Comment:
why you need the database instance as a future?

Reply:
i was doing that because i needed to await the db connection somewhere, and i was awaiting it every time the getData() method was called. Now i do it on a singleton dependencies class, and i receive on the repositories a AppDatabase instead of a Future<AppDatabase>